### DISCUSSION TITLE
Unity Editor can’t find dependency DLLs in subfolders anymore (DLLNotFoundException)

### DISCUSSION THREAD

**Discussion by SashelI** [mono/SkiaSharp#3023] [Q&A]
### Description

Hi,

I came into some frustrating issue today in the editor : I have a project in which I have manually imported libraries. All of the required dependencies are also in the project, in a subfolder (see screenshots below).

This project was working fine and finding everything it needed. Today I downloaded the new version of the libraries (from skiasharp 2.88.7 to skiasharp 2.88.8), replaced DLLs without changing any of the folder structure, and suddendly, with two different libraries _(libSkiaSharp and pdfium from SkiaSharp and PDFToImage)_, when I hit play i have a “DLLNotFoundException” on the dependencies, but **only** if they are in a subfolder…

I have many dlls and .so because this is a cross-platform project (android ARM64 + android x64 + UWP ARM64 + Winx64), so putting every single dependency dll in the same folder is a no-go.

Anyone knows why this happens ?
Thanks.

Here are the screenshots :

**Working old project structure, this exact same structure on the updated dlls doesn’t work, and I have to put libskiasharp.dll in the root Skia folder :**

![image](https://github.com/user-attachments/assets/f292ebb3-c39a-4cca-96a6-aeadad510bc9)
![image](https://github.com/user-attachments/assets/259e8a39-a92c-45e6-954b-90693116a401)

_(And same goes for pdfium.dll)_

### Code

[https://github.com/SashelI/skiadll-tests-unity](https://github.com/SashelI/skiadll-tests-unity)

### Actual Behavior

`DllNotFoundException: libSkiaSharp assembly:<unknown assembly> type:<unknown type> member:(null)
SkiaSharp.SKImageInfo..cctor () (at <ba5db125678d4cb8bfb461a6f3fca1c2>:0)
Rethrow as TypeInitializationException: The type initializer for 'SkiaSharp.SKImageInfo' threw an exception.`

### Version of SkiaSharp

Other (Please indicate in the description)

### Last Known Good Version of SkiaSharp

Other (Please indicate in the description)

### IDE / Editor

Other (Please indicate in the description)

### Platform / Operating System

Windows

### Platform / Operating System Version

Windows 10 x64

### Relevant Screenshots

![image](https://github.com/user-attachments/assets/8888366f-ab43-4256-b082-9a4e3bc2c252)
![image](https://github.com/user-attachments/assets/4990cd5f-0490-446b-a0b8-4a9d54a68bec)

### Relevant Log Output

```shell
`DllNotFoundException: libSkiaSharp assembly:<unknown assembly> type:<unknown type> member:(null)
SkiaSharp.SKImageInfo..cctor () (at <ba5db125678d4cb8bfb461a6f3fca1c2>:0)
Rethrow as TypeInitializationException: The type initializer for 'SkiaSharp.SKImageInfo' threw an exception.`
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by SashelI** [comment]
Update : the issue is even weirder : it actually can’t find the dll anywhere on each project startup until I move the dll file to a different folder ?? But no matter the folder :

I have my dll in a Skia/win-64/native subfolder.
I open the Unity project, click play : DLLNotFound
I ctrl+x the dll, put it in Skia/ folder.
hit play, it works
I close the editor
I open the Unity project again
I hit play : DLLNotFound
I ctrl+x the dll, put it in Skia/win-64/native subfolder
Hit play, it works
What ?

Here is a repo with the issue on it, if anyone wants to give it a try : [https://github.com/SashelI/skiadll-tests-unity](https://github.com/SashelI/skiadll-tests-unity)

**Reply by mattleibow** [reply]
Could this be a bug in Unity? I am not very familiar with Unity, so maybe this is better a question on Stack Overflow or their forums?

The fact that it works the first time means that SkiaSharp is working. Breaking when restarting the editor does seem to be something that SkiaSharp does not have any control over or even should be influenced by.

**Reply by SashelI** [reply]
Hi, I have posted the question on both forums (here and on [Unity Discussions](https://discussions.unity.com/t/editor-cant-find-dependency-dlls-in-subfolders-anymore/1524348/7)) at the same time. I haven't got a probant answer yet, so I don't know where it comes from tbh.
